Token.js

```diff
@@ -1074,3 +1043,39 @@ function deselect_all_tokens() {
 		}
 	}
 }
+
+function token_health_aura(hpPercentage) {
+	//PERC TO RGB------------
+	const percentToHEX = function (percent) {
+		var HEX;
+		if (percent > 100) HEX = "#0000FF";
+		else {
+			if (percent === 100) percent = 99;
+			var r, g, b = 0;
+			if (percent < 50) {
+				g = Math.floor(255 * (percent / 50));
+				r = 255;
+			}
+			else {
+				g = 255;
+				r = Math.floor(255 * ((50 - percent % 50) / 50));
+			}
+			HEX = "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+		}
+		return HEX;
+	}
+	//HEX TO RGB------------
+	const hexToRGB = function (hex) {
+		// Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+		var shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
+		hex = hex.replace(shorthandRegex, function (m, r, g, b) {
+			return r + r + g + g + b + b;
+		});
+
+		const pHex = (n) => parseInt(n, 16);
+
+		var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+		return result ? `rgb(${pHex(result[1])} ${pHex(result[2])} ${pHex(result[3])} / 80%)` : null;
+	}
+	return hexToRGB(percentToHEX(hpPercentage));
+}

```

Comment:
QUESTION

Why are we converting HEX to RGB? 
I css both are accepted values `rgb(255, 255, 255)` === `#ffffff`.

Or am I missing something? 🤔 

And if you need colours in percentage we could even use the HSL colour notation: 
- H hue
- S Saturation
- L Lightness
https://www.w3schools.com/css/css_colors_hsl.asp

Reply:
wouldn't we need also a function to convert it from HEX to HSL?

Reply:
HEX, HSL, and RGB are all valid color declarations in css. 
So, why do we need to convert them in one way or another?

Reply:
Search for `CSS Color Values` in the page: https://www.w3schools.com/csS/css_colors.asp

Reply:
because I need to set opacity on the color